frontend/src/bbkit/BBSelect.vue

```diff
@@ -1,148 +1,109 @@
-                $emit('select-item', item, () => {
-                  state.selectedItem = item;
-                });
-              }
-              close();
-            "
+          <heroicons-solid:selector class="h-5 w-5 text-control-light" />
+        </span>
+      </button>
+    </VTarget>
+    <VFollower :show="state.showMenu" placement="bottom-start">
+      <transition :appear="true" name="fade-fast">
+        <div
+          v-if="state.showMenu"
+          ref="popup"
+          class="z-50 rounded-md bg-white shadow-lg"
+          :style="{ width: `${width}px` }"
+        >
+          <ul
+            tabindex="-1"
+            role="listbox"
+            aria-labelledby="listbox-label"
+            aria-activedescendant="listbox-item-3"
+            class="max-h-56 rounded-md py-1 ring-1 ring-black ring-opacity-5 overflow-auto focus:outline-none sm:text-sm"
           >
-            <slot name="menuItem" :item="item" />
-            <span
-              v-if="item === state.selectedItem"
-              class="absolute inset-y-0 right-0 flex items-center pr-4"
+            <!--
+              Select option, manage highlight styles based on mouseenter/mouseleave and keyboard navigation.
+              Highlighted: "text-white bg-indigo-600", Not Highlighted: "text-gray-900"
+            -->
+            <li
+              v-for="(item, index) in itemList"
+              :key="index"
+              role="option"
+              class="text-main hover:text-main-text hover:bg-main-hover cursor-default select-none relative py-2 pl-3 pr-9"
+              @click="
+                if (item !== state.selectedItem) {
+                  $emit('select-item', item, () => {
+                    state.selectedItem = item;
+                  });
+                }
+                close();
+              "
             >
-              <!-- Heroicon name: solid/check -->
-              <heroicons-solid:check class="h-5 w-5" />
-            </span>
-          </li>
-        </ul>
-      </div>
-    </transition>
-  </div>
+              <slot name="menuItem" :item="item" />
+              <span
+                v-if="item === state.selectedItem"
+                class="absolute inset-y-0 right-0 flex items-center pr-4"
+              >
+                <!-- Heroicon name: solid/check -->
+                <heroicons-solid:check class="h-5 w-5" />
+              </span>
+            </li>
+          </ul>
+        </div>
+      </transition>
+    </VFollower>
+  </VBinder>
 </template>
 
 <script lang="ts">
-import { reactive, PropType, watch } from "vue";
-import clickInsideOutside from "./directives/click-inside-outside";
+import { reactive, PropType, watch, defineComponent, ref } from "vue";
+import { VBinder, VTarget, VFollower } from "vueuc";
+import { onClickOutside, useElementBounding } from "@vueuse/core";
 
 interface LocalState {
   showMenu: boolean;
   selectedItem: any;
 }
 
-export default {
+type ItemType = any;
```

Comment:
Do we need define the Type of Option, something like

```ts
{ label: string, value: string }
```

Reply:
> Do we need define the Type of Option, something like
> 
> ```ts
> { label: string, value: string }
> ```

It's now `string | number | any` in real use cases.
Changing it to `{ label: string, value: any }` is good to match `<NSelect>` and `<NPopselect>` but will break other components (need a refactor of type conversion).

Reply:
If the refactor of type conversion will take you a lot of time on it, just keep `any` for now. Maybe refactor it in the future.

Reply:
I'll change it to `string | number | any` for now.